yb-voyager/src/tgtdb/event.go

```diff
@@ -257,6 +257,12 @@ type EventBatch struct {
 	EventCountsByTable map[string]*EventCounter
 }
 
+func NewEventBatch(events []*Event, chanNo int, targetSchema string) *EventBatch {
+	batch := &EventBatch{Events: events, ChanNo: chanNo, EventCounts : &EventCounter{}, EventCountsByTable: make(map[string]*EventCounter)}
+	batch.EventCountsByTable = batch.getEventCountsByTable(targetSchema)
```

Comment:
In a single loop you should update both `EventCounts` and `EventCountsByTable`.

Reply:
Yes, that is happening in `getEventCountsByTable()` func - 
```
func (eb *EventBatch) getEventCountsByTable(targetSchema string) map[string]*EventCounter {
	counts := make(map[string]*EventCounter)
	for _, event := range eb.Events {
		tableName := event.getTableName(targetSchema)
		if _, ok := counts[tableName]; !ok {
			counts[tableName] = &EventCounter{}
		}
		counts[tableName].CountEvent(event)
		eb.EventCounts.CountEvent(event)
	}
	return counts
}
```